File: Anush_Ghazaryan/Homeworks/Python/2_12.06/calculator.py

import math
def Add(num1,num2):
    res=num1+num2
    print('%d + %d = %d ' %( num1, num2, res))


def Sub(num1,num2):
    res=num1-num2
    print('%d - %d = %d ' %( num1, num2, res))
    

def Mul(num1,num2):
    res=num1*num2
    print('%d * %d = %d ' %( num1, num2, res))


def Div(num1,num2):
    if num2 != 0 :
        res=num1/num2
        print('%d / %d = %d ' %( num1, num2, res))

    else:
        print("The Number Cannot be divided by 0")

def Pow(num1,num2):
    res=math.pow(num1,num2)
    print('%d ^ %d = %d ' %( num1, num2, res))


def calc():
    num1 = int(input('Input 1st Number: '))
    num2 = int(input('Input 2nd Number: '))
    print("Operations")
    print("1. + ")
    print("2. - ")
    print("3. * ")
    print("4. / ")
    print("5. ^ ")
    try :

        op = input('Choose an operation: ')
    except NameError :
	print("There is no such an operator.Try again!")

    if op == 1 :
        Add(num1,num2)
    elif op == 2 :
	Sub(num1,num2)
    elif op == 3 :
	Mul(num1,num2)
    elif op == 4 :
	    Div(num1,num2)
    elif op == 5 :
        Pow(num1,num2)
def main():
    calc()
    
main()

File: Arman_Avdalyan/Homeworks/Java/6_12.07/CalcGeneric/Calculator.java
public class Calculator<T extends Number> {

    public Number add(T num1, T num2) {
        String type = num1.getClass().getName();
        if (type == "java.lang.Integer") {
            return (Integer)num1 + (Integer)num2;
        }
        if (type == "java.lang.Double") {
            return (Double)num1 + (Double) num2;
        }
        if (type == "java.lang.Float") {
            return (Float)num1 + (Float) num2;
        }
        return null;
    }

    public Number sub(T num1, T num2) {
        String type = num1.getClass().getName();
        if (type == "java.lang.Integer") {
            return (Integer)num1 - (Integer)num2;
        }
        if (type == "java.lang.Double") {
            return (Double)num1 - (Double) num2;
        }
        if (type == "java.lang.Float") {
            return (Float)num1 - (Float) num2;
        }
        return null;
    }

    public Number div(T num1, T num2) {
        String type = num1.getClass().getName();
        if (type == "java.lang.Integer") {
            return (Integer)num1 / (Integer)num2;
        }
        if (type == "java.lang.Double") {
            return (Double)num1 / (Double) num2;
        }
        if (type == "java.lang.Float") {
            return (Float)num1 / (Float) num2;
        }
        return null;
    }

    public Number mul(T num1, T num2) {
        String type = num1.getClass().getName();
        if (type == "java.lang.Integer") {
            return (Integer)num1 * (Integer)num2;
        }
        if (type == "java.lang.Double") {
            return (Double)num1 * (Double) num2;
        }
        if (type == "java.lang.Float") {
            return (Float)num1 * (Float) num2;
        }
        return null;
    }
}